docs/secure_coding_and_deployment_hardening/atom_exhaustion.md

```diff
@@ -8,6 +8,8 @@ next:
   title: Serialisation and deserialisation
 ---
 
+* When converting lists or binaries to atoms, and the set of desired atoms is known, consider converting them yourself with a lookup table instead of using the general functions for doing so (e.g. [list_to_existing_atom/1](https://erlang.org/doc/man/erlang.html#list_to_existing_atom-1) or [List.to_existing_atom/1](https://hexdocs.pm/elixir/List.html#to_existing_atom/1)). This not only prevents exhaustion but also ensures that undesired values do not slink through.
```

Comment:
What do you think about this suggestion?

```suggestion
* When converting strings or binaries to atoms, it’s crucial to ensure only a predefined set of atoms is used.
  Instead of using functions like [`list_to_existing_atom/1`](https://erlang.org/doc/man/erlang.html#list_to_existing_atom-1) or
  [`List.to_existing_atom/1`](https://hexdocs.pm/elixir/List.html#to_existing_atom/1), which can still accept
  arbitrary input and introduce unintended atoms, it's safer to maintain a lookup table that maps known strings
  to atoms. This prevents the risk of converting unexpected or harmful values, such as an existing module name,
  into atoms. Using this method not only safeguards against atom table exhaustion but also ensures strict control
  over which atoms are allowed in your application.
```

Reply:
I like it, let's go with that :-)